Panic with krakend-opencensus

When adding **github_com/devopsfaith/krakend-opencensus** in global **extra_config** with **kakrend-ce v1.40**. We get the following error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1971650]

goroutine 1 [running]:
github.com/devopsfaith/krakend-opencensus.GetAggregatedPathForBackendMetrics(0x0, 0x3c79000)
	/go/pkg/mod/github.com/devopsfaith/krakend-opencensus@v1.4.0/opencensus.go:396 +0x90
github.com/devopsfaith/krakend-opencensus.HTTPRequestExecutorFromConfig(0x28d4768, 0x0, 0x287ca82)
	/go/pkg/mod/github.com/devopsfaith/krakend-opencensus@v1.4.0/http.go:31 +0x38
github.com/devopsfaith/krakend-opencensus.HTTPRequestExecutor(...)
	/go/pkg/mod/github.com/devopsfaith/krakend-opencensus@v1.4.0/http.go:24
github.com/devopsfaith/krakend-ce.NewBackendFactoryWithContext.func1(0xc000162dc0, 0xc00023dec0)
	/app/backend_factory.go:50 +0x85
github.com/luraproject/lura/transport/http/client/plugin.HTTPRequestExecutor.func1(0xc000162dc0, 0xc0004f8e10)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/transport/http/client/plugin/executor.go:24 +0x978
github.com/devopsfaith/krakend-martian.NewConfiguredBackendFactory.func1(0xc000162dc0, 0x2ded028)
	/go/pkg/mod/github.com/devopsfaith/krakend-martian@v1.4.0/martian.go:42 +0x48
github.com/devopsfaith/krakend-pubsub.(*BackendFactory).New(0xc0000b64e0, 0xc000162dc0, 0x0)
	/go/pkg/mod/github.com/devopsfaith/krakend-pubsub@v1.4.0/client.go:55 +0xa9
github.com/devopsfaith/krakend-amqp.backendFactory.New(0x2ded028, 0xc0006684c0, 0x2dfa940, 0xc0008f2300, 0xc0004f8240, 0xc0000b6510, 0xc0006c0340, 0xc000162dc0, 0x601fafaaf37b86d5)
	/go/pkg/mod/github.com/devopsfaith/krakend-amqp@v1.4.0/client.go:43 +0x10b
github.com/devopsfaith/krakend-lambda.BackendFactoryWithInvoker.func1(0xc000162dc0, 0xc00093e738)
	/go/pkg/mod/github.com/devopsfaith/krakend-lambda@v1.4.0/backend.go:51 +0x15a
github.com/devopsfaith/krakend-cel.BackendFactory.func1(0xc000162dc0, 0x2697a00)
	/go/pkg/mod/github.com/devopsfaith/krakend-cel@v1.4.0/proxy.go:41 +0x59
github.com/devopsfaith/krakend-lua/proxy.BackendFactory.func1(0xc000162dc0, 0x28d4680)
	/go/pkg/mod/github.com/devopsfaith/krakend-lua@v1.4.0/proxy/proxy.go:38 +0x5f
github.com/devopsfaith/krakend-ratelimit/juju/proxy.BackendFactory.func1(0xc000162dc0, 0x2dfa940)
	/go/pkg/mod/github.com/devopsfaith/krakend-ratelimit@v1.4.0/juju/proxy/proxy.go:59 +0x55
github.com/devopsfaith/krakend-circuitbreaker/gobreaker/proxy.BackendFactory.func1(0xc000162dc0, 0x7)
	/go/pkg/mod/github.com/devopsfaith/krakend-circuitbreaker@v1.4.0/gobreaker/proxy/proxy.go:38 +0x67
github.com/devopsfaith/krakend-metrics.(*Metrics).BackendFactory.func1(0xc000162dc0, 0xe)
	/go/pkg/mod/github.com/devopsfaith/krakend-metrics@v1.4.0/proxy.go:40 +0x82
github.com/devopsfaith/krakend-opencensus.BackendFactory.func1(0xc000162dc0, 0x3b5e588)
	/go/pkg/mod/github.com/devopsfaith/krakend-opencensus@v1.4.0/proxy.go:62 +0x9a
github.com/luraproject/lura/proxy.defaultFactory.newStack(0xc0004f8320, 0x2dfa940, 0xc0008f2300, 0x28d4748, 0xc000162dc0, 0x8)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:87 +0x34
github.com/luraproject/lura/proxy.defaultFactory.newSingle(...)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:83
github.com/luraproject/lura/proxy.defaultFactory.New(0xc0004f8320, 0x2dfa940, 0xc0008f2300, 0x28d4748, 0xc000985900, 0xc0001988f0, 0xc000132af0, 0xc00057a0c0)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:59 +0x236
github.com/luraproject/lura/proxy.shadowFactory.New(0x2dc7660, 0xc00023dd80, 0xc000985900, 0xd5afdc, 0xc000000180, 0x300000002)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/shadow.go:39 +0x34f
github.com/devopsfaith/krakend-jsonschema.ProxyFactory.func1(0xc000985900, 0xc0005ae2d0, 0x71, 0xc00093ed08)
	/go/pkg/mod/github.com/devopsfaith/krakend-jsonschema@v1.4.0/jsonschema.go:26 +0x3d
github.com/luraproject/lura/proxy.FactoryFunc.New(0xc000567d70, 0xc000985900, 0xc00093eda8, 0xe03f53, 0xc000134060)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:24 +0x30
github.com/devopsfaith/krakend-cel.ProxyFactory.func1(0xc000985900, 0xc0005ae200, 0x71, 0x90)
	/go/pkg/mod/github.com/devopsfaith/krakend-cel@v1.4.0/proxy.go:17 +0x63
github.com/luraproject/lura/proxy.FactoryFunc.New(0xc0000b6570, 0xc000985900, 0x0, 0x3b199e1f00000002, 0x71)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:24 +0x30
github.com/devopsfaith/krakend-lua/proxy.ProxyFactory.func1(0xc000985900, 0x8, 0x9, 0x10000c0006c0380)
	/go/pkg/mod/github.com/devopsfaith/krakend-lua@v1.4.0/proxy/proxy.go:21 +0x69
github.com/luraproject/lura/proxy.FactoryFunc.New(0xc0000b65a0, 0xc000985900, 0x203000, 0x203000, 0x0)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:24 +0x30
github.com/devopsfaith/krakend-metrics.(*Metrics).ProxyFactory.func1(0xc000985900, 0x203000, 0x203000, 0x203000)
	/go/pkg/mod/github.com/devopsfaith/krakend-metrics@v1.4.0/proxy.go:26 +0x58
github.com/luraproject/lura/proxy.FactoryFunc.New(0xc0000b65d0, 0xc000985900, 0x9, 0xc00093f0b0, 0xd25ffb)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:24 +0x30
github.com/luraproject/lura/proxy.FactoryFunc.New(0xc000567d88, 0xc000985900, 0x40, 0xc000668c00, 0x2)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/proxy/factory.go:24 +0x30
github.com/luraproject/lura/router/gin.ginRouter.registerKrakendEndpoints(0xc000123a00, 0x0, 0x0, 0x0, 0xc00023dde0, 0x2dc7640, 0xc000567d88, 0x2dfa940, 0xc0008f2300, 0xc00023de60, ...)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/router/gin/router.go:117 +0x8b
github.com/luraproject/lura/router/gin.ginRouter.Run(0xc000123a00, 0x0, 0x0, 0x0, 0xc00023dde0, 0x2dc7640, 0xc000567d88, 0x2dfa940, 0xc0008f2300, 0xc00023de60, ...)
	/go/pkg/mod/github.com/luraproject/lura@v1.4.0/router/gin/router.go:106 +0x312
github.com/devopsfaith/krakend-ce.(*ExecutorBuilder).NewCmdExecutor.func1(0xc0006320c0, 0xd, 0xc000132560, 0x1, 0x1, 0xb2d05e00, 0x0, 0x3c76490, 0x0, 0x0, ...)
	/app/executor.go:171 +0x835
github.com/devopsfaith/krakend-cobra.runFunc(0x3af7a60, 0xc0004bc780, 0x0, 0x2)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v1.4.0/run.go:24 +0x1df
github.com/spf13/cobra.(*Command).execute(0x3af7a60, 0xc0004bc4c0, 0x2, 0x2, 0x3af7a60, 0xc0004bc4c0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x3af7560, 0xc00086aae0, 0x28d8118, 0xc00086aae4)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/devopsfaith/krakend-cobra.Root.Execute(0x3af7560, 0xc00082d450, 0x2, 0x2, 0xc00086aab0, 0xc00052b770, 0x2, 0x2, 0xc00086aae0, 0x2dc7340, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v1.4.0/cmd.go:119 +0x69
github.com/devopsfaith/krakend-cobra.ExecuteRoot(0x2dc7340, 0xc00057e120, 0xc0003f8640, 0x3af7560, 0xc00082d450, 0x2, 0x2, 0xc00086aab0, 0xc00052b770, 0x2, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v1.4.0/cmd.go:26 +0x98
github.com/devopsfaith/krakend-cobra.Execute(...)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v1.4.0/cmd.go:21
main.main()
	/app/cmd/krakend-ce/main.go:56 +0x3ae
```

--- a/opencensus.go
+++ b/opencensus.go
@@ -337,6 +337,10 @@ var replaceMetricBackendPath = regexp.MustCompile(`{{.(.*?)}}`)
 
 // GetAggregatedPathForMetrics returns a path aggregator function ready to reduce path cardinality in the metrics
 func GetAggregatedPathForMetrics(cfg *config.EndpointConfig) func(r *http.Request) string {
+	if cfg == nil {
+		return simplePathExtractor
+	}
+
 	aggregationMode := aggregationModePattern
 	endpointExtraCfg, endpointExtraCfgErr := parseEndpointConfig(cfg)
 	if endpointExtraCfgErr == nil {
@@ -367,6 +371,9 @@ func GetAggregatedPathForMetrics(cfg *config.EndpointConfig) func(r *http.Reques
 
 // GetAggregatedPathForBackendMetrics returns a path aggregator function ready to reduce path cardinality in the metrics
 func GetAggregatedPathForBackendMetrics(cfg *config.Backend) func(r *http.Request) string {
+	if cfg == nil {
+		return simplePathExtractor
+	}
 	aggregationMode := aggregationModePattern
 	endpointExtraCfg, endpointExtraCfgErr := parseBackendConfig(cfg)
 	if endpointExtraCfgErr == nil {

--- a/opencensus_test.go
+++ b/opencensus_test.go
@@ -9,15 +9,15 @@ import (
 
 func TestGetAggregatedPathForMetrics(t *testing.T) {
 	for i, tc := range []struct {
-		cfg      config.EndpointConfig
+		cfg      *config.EndpointConfig
 		expected string
 	}{
 		{
-			cfg:      config.EndpointConfig{Endpoint: "/api/:foo/:bar"},
+			cfg:      &config.EndpointConfig{Endpoint: "/api/:foo/:bar"},
 			expected: "/api/{foo}/{bar}",
 		},
 		{
-			cfg: config.EndpointConfig{
+			cfg: &config.EndpointConfig{
 				Endpoint: "/api/:foo/:bar",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "pattern"},
@@ -26,7 +26,7 @@ func TestGetAggregatedPathForMetrics(t *testing.T) {
 			expected: "/api/{foo}/{bar}",
 		},
 		{
-			cfg: config.EndpointConfig{
+			cfg: &config.EndpointConfig{
 				Endpoint: "/api/:foo/:bar",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "lastparam"},
@@ -35,7 +35,7 @@ func TestGetAggregatedPathForMetrics(t *testing.T) {
 			expected: "/api/foo/{bar}",
 		},
 		{
-			cfg: config.EndpointConfig{
+			cfg: &config.EndpointConfig{
 				Endpoint: "/api/:foo/:bar",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "off"},
@@ -43,8 +43,11 @@ func TestGetAggregatedPathForMetrics(t *testing.T) {
 			},
 			expected: "/api/foo/bar",
 		},
+		{
+			expected: "/api/foo/bar",
+		},
 	} {
-		extractor := GetAggregatedPathForMetrics(&tc.cfg)
+		extractor := GetAggregatedPathForMetrics(tc.cfg)
 		r, _ := http.NewRequest("GET", "http://example.tld/api/foo/bar", nil)
 		if tag := extractor(r); tag != tc.expected {
 			t.Errorf("tc-%d: unexpected result: %s", i, tag)
@@ -54,15 +57,15 @@ func TestGetAggregatedPathForMetrics(t *testing.T) {
 
 func TestGetAggregatedPathForBackendMetrics(t *testing.T) {
 	for i, tc := range []struct {
-		cfg      config.Backend
+		cfg      *config.Backend
 		expected string
 	}{
 		{
-			cfg:      config.Backend{URLPattern: "/api/{{.Foo}}/{{.Bar}}"},
+			cfg:      &config.Backend{URLPattern: "/api/{{.Foo}}/{{.Bar}}"},
 			expected: "/api/{foo}/{bar}",
 		},
 		{
-			cfg: config.Backend{
+			cfg: &config.Backend{
 				URLPattern: "/api/{{.Foo}}/{{.Bar}}",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "pattern"},
@@ -71,7 +74,7 @@ func TestGetAggregatedPathForBackendMetrics(t *testing.T) {
 			expected: "/api/{foo}/{bar}",
 		},
 		{
-			cfg: config.Backend{
+			cfg: &config.Backend{
 				URLPattern: "/api/{{.Foo}}/{{.Bar}}",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "lastparam"},
@@ -80,7 +83,7 @@ func TestGetAggregatedPathForBackendMetrics(t *testing.T) {
 			expected: "/api/foo/{bar}",
 		},
 		{
-			cfg: config.Backend{
+			cfg: &config.Backend{
 				URLPattern: "/api/{{.Foo}}/{{.Bar}}",
 				ExtraConfig: config.ExtraConfig{
 					Namespace: map[string]interface{}{"path_aggregation": "off"},
@@ -88,8 +91,11 @@ func TestGetAggregatedPathForBackendMetrics(t *testing.T) {
 			},
 			expected: "/api/foo/bar",
 		},
+		{
+			expected: "/api/foo/bar",
+		},
 	} {
-		extractor := GetAggregatedPathForBackendMetrics(&tc.cfg)
+		extractor := GetAggregatedPathForBackendMetrics(tc.cfg)
 		r, _ := http.NewRequest("GET", "http://example.tld/api/foo/bar", nil)
 		if tag := extractor(r); tag != tc.expected {
 			t.Errorf("tc-%d: unexpected result: %s", i, tag)

